src/main/java/core/org/akaza/openclinica/dao/hibernate/StudyDao.java

```diff
@@ -420,15 +420,17 @@ public HashMap getChildrenByParentIds(ArrayList allStudies) {
         return null;
     }
     @Transactional
-    public Study updateSitesStatus(Study s) {
+    public void updateSitesStatus(Study s) {
         String query = "from "+ getDomainClassName() +" s WHERE s.study.studyId = :parentStudyId";
         Query q=getCurrentSession().createQuery(query);
         q.setParameter("parentStudyId", s.getStudyId());
-        Study s1 = (Study) q.uniqueResult();
-        s1.setStatus(s.getStatus());
-        s1.setOldStatusId(s.getOldStatusId());
-        getCurrentSession().update(s1);
-        return s1;
+        List<Study> slist = q.getResultList();
+        
+        for(Study s1:slist) {
+			s1.setStatus(s.getStatus());
+			s1.setOldStatusId(s.getOldStatusId());
+			getCurrentSession().update(s1);
+        }
     }
```

Comment:
Instead of running select and update, create an update query to update all sites with one query. See the jdbc query for example.

Reply:
done